Feature: datepicker

rolandoldengarm opened the pull request:
The datepicker implementation. Fixes https://github.com/ngOfficeUIFabric/ng-officeuifabric/issues/19

Issues from first PR have been resolved https://github.com/ngOfficeUIFabric/ng-officeuifabric/pull/162

jjczopek committed:
fix(spinner): spinner sizing & positioning for Fabric UI 2.0.1

Due to change in Fabric UI CSS regarding label size of spinner, sizing and placement logic of the spinner has changed. This has been aligned with the changes.

Closes #177.

andrewconnell committed:
feat(): update to Office UI Fabric v2.0.1

Updated the dependency from Office UI Fabric v1.2.1 to v2.0.1. This fixes numerous bugs ngOfficeUiFabric found in Office UI Fabric that are listed here: https://github.com/ngOfficeUIFabric/ng-officeuifabric/blob/master/docs/guides/FAQ.md.

Closes #167.

andrewconnell committed:
chore(): remove npm log file

NPM log file mistakenly added to source when committing the button directive in 0550008968e60ac958eea8604f05d6b2d92574bf

[ci skip]

andrewconnell committed:
docs(): update FAQ

Included Office UI Fabric bug https://github.com/OfficeDev/Office-UI-Fabric/issues/325 in list of bugs we've found.

[ci skip]

andrewconnell committed:
docs(): update FAQ for Fabric bug #328

Update FAQ to reference the Office UI Fabric issue https://github.com/OfficeDev/Office-UI-Fabric/issues/328 found by @jjczopek.

[ci skip]

Commit:
Initial check in of DatePicker component

s-KaiNet committed:
chore(): update gulp test task

Updated gulp **test** task to have a dependency on **transpile-ts**.

Closes #180. Closes #181.

Commit:
Update demo

andrewconnell committed:
docs(release): 0.4.0

Commit:
Fixed the month picker. Year picker still work in progress.
Demo updates.

Commit:
Additional test scripts
year and month picker improvements
package.json to include pickadate@3.5.3 (highest supported version at this stage)

Commit:
Code coverage increased: Additional tests and removed unused code.

Commit:
Error in package.json: Version must match 3.5.3 exactly. ^ will install latest minor version.

andrewconnell commented:
### Update Attribute Name?

For the `uif-placeholder-text` attribute, why not use `placeholder` since that's a well known HTML attribute and you're doing the same thing with the custom one?

### Failing Test

One test is failing though locally for me. I see it passed all tests in travis, but i’ve blown away all compiled JS & `node_modules`, pulling down everything fresh and rebuilding, but still get one failing test.

![image](https://cloud.githubusercontent.com/assets/2068657/12945710/3a114c58-cfbd-11e5-814a-c330747e87cf.png)

I noticed your branch on your repo is 22 commits behind `master`. I pulled your branch down and ran it… all tests passed. I then cloned a clean copy of the main repo & pulled in your PR… same one failing test.

I’m not sure why it’s passing on travis but failing here… can you take a look?

Like I said, I suggest you rebase your local branch off a updated `dev` and run your tests… also suggest you clean out all JS `gulp clean` and blow away your `node_modules` folder and do a clean `npm install` to make sure you’ve eliminated those variances.

andrewconnell committed:
docs(faq): added bug in fabric found in datepicker directive

When creating `uif-datepicker` directive, bug found in Office UI Fabric v2.0.1 when using latest version of `pickadate.js`. Reported as https://github.com/OfficeDev/Office-UI-Fabric/issues/336.

[ci skip]

Commit:
Renamed uif-placeholder-text to placeholder.
The demoBasicUsage had an old, wrong demo; fixed.

Commit:
Replaced uif-placeholder-text with placeholder

Commit:
Localization fixes

The unit tests were not working on time zones on the left of UTC.
